protected/humhub/modules/admin/controllers/InformationController.php

```diff
@@ -78,15 +84,32 @@ public function actionAbout()
 
     public function actionPrerequisites()
     {
-        return $this->render('prerequisites', ['checks' => \humhub\libs\SelfTest::getResults()]);
+        return $this->render('prerequisites', ['checks' => SelfTest::getResults()]);
     }
 
-    public function actionDatabase()
+    public function actionDatabase(int $migrate = self::DB_ACTION_CHECK)
     {
+
+        $request = Yii::$app->request;
+
+        if ($migrate === self::DB_ACTION_RUN) {
+            $migrationOutput = sprintf(
+                "%s\n%s",
+                MigrateController::webMigrateAll(),
+                SettingController::flushCache()
+            );
+        } else {
+            $migrationOutput = MigrateController::webMigrateAll(MigrateController::MIGRATION_ACTION_NEW);
+
+            $migrate = str_contains($migrationOutput, 'No new migrations found.')
+                ? self::DB_ACTION_CHECK
+                : self::DB_ACTION_PENDING;
+        }
+
         $databaseInfo = new DatabaseInfo(Yii::$app->db->dsn);
 
         $rebuildSearchJob = new RebuildIndex();
-        if (Yii::$app->request->isPost && Yii::$app->request->get('rebuildSearch') == 1) {
+        if ($request->isPost && $request->get('rebuildSearch') == 1) {
```

Comment:
Just a small note: I understand that we often have very deep wiring that is a bit hard to read, but in the case of `Yii::$app->request` or other app components I would like to keep it that way. :-)

Reply:
If `Yii::$app->request` was an actual class field, I'd understand. And I agree that in this case line 93 could be at 112 to keep it more readable.

But what is be the benefit of going through the whole get-component thing twice, just to read two properties of the very same object? Isn't that a bunch of function calls we can safe?

I appreciate that this should not be done if the object might change in between, e.g. `Yii::$app->controller` but in this case I just make to successive calls to `isPost` and `get`.

Shall I undo the change or move the $request asignment to line 112?

Reply:
I would like to undo this, I don't want to be too picky.

I believe the Yii2 ServiceLocator is quite efficient: 
https://github.com/yiisoft/yii2/blob/master/framework/di/ServiceLocator.php#L127-L129

Reply:
done.

Reply:
> I believe the Yii2 ServiceLocator is quite efficient:
> https://github.com/yiisoft/yii2/blob/master/framework/di/ServiceLocator.php#L127-L129

Btw, internally it's more something like:
1. Application: property_exists('request')
3. Application: get_parent_class()
4. WebApplication: property_exists('request')
3. WebApplication: get_parent_class()
4. BaseApplication: property_exists('request')
3. BaseApplication: get_parent_class()
4. Module: property_exists('request')
3. Module: get_parent_class()
4. ServiceLocator: property_exists('request')
3. ServiceLocator: get_parent_class()
4. Component: property_exists('request')
3. Component: get_parent_class()
4. BaseObject: property_exists('request')
3. BaseObject: get_parent_class()
5. Application: method_exists('__get')
3. Application: get_parent_class()
4. WebApplication: method_exists('__get')
3. WebApplication: get_parent_class()
4. BaseApplication: method_exists('__get')
3. BaseApplication: get_parent_class()
4. Module: method_exists('__get')
3. Module: get_parent_class()
4. ServiceLocator: method_exists('__get')
4. ServiceLocator::__get('request')
6. Application: method_exists('has')
3. Application: get_parent_class()
4. WebApplication: method_exists('has')
3. WebApplication: get_parent_class()
4. BaseApplication: method_exists('has')
3. BaseApplication: get_parent_class()
4. Module: method_exists('has')
3. Module: get_parent_class()
4. ServiceLocator: method_exists('has')
5. isset($this->_components['has']) with array lookup
6. return true
6. Application: method_exists('get')
3. Application: get_parent_class()
4. WebApplication: method_exists('get')
3. WebApplication: get_parent_class()
4. BaseApplication: method_exists('get')
3. BaseApplication: get_parent_class()
4. Module: method_exists('get')
3. Module: get_parent_class()
4. ServiceLocator: method_exists('get')
5. ServiceLocator::get('request')
5. isset($this->_components['has']) with array lookup
5. return $this->_components['has'] with array lookup

It may be though, that the first location of the property (steps 1-14) are more efficient for an instantiated object. I have not checked the source code. But if it's similar as reflection works, that would basically be the procedure.

And all of that twice in the current method. Only once, if it's not a post request. :-)

Using `Yii::$app->get('request')` would already be massively better. But then we loose autocomplete.

So I certainly think, storing the variable is way more efficient. Particularly if you do it many times in the same scope, and then on many levels ...

Reply:
Thanks for the overview, dood to know, indeed, it doesn't look very efficient. Hope there are some Optimizations in the background. :-)

But for me and for now, the important thing is to maintain a consistent style. But for sure there is much room for improvements in future